src/boosting/bagging.hpp

```diff
@@ -0,0 +1,181 @@
+/*!
+ * Copyright (c) 2021 Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE file in the project root for license information.
+ */
+
+#ifndef LIGHTGBM_SAMPLE_STRATEGY_BAGGING_HPP_
+#define LIGHTGBM_SAMPLE_STRATEGY_BAGGING_HPP_
+
+namespace LightGBM {
+
+class BAGGING : public SampleStrategy {
```

Comment:
Otherwise the constructor will have the same name as `Bagging` method.

Reply:
I'll change the class name into `BaggingSampleStrategy`.

Reply:
Thanks, works for me!

I think `GOSS` -> `GOSSStrategy` renaming will be good as well.
P.S. `GOSS` all caps here because it's abbreviation of `Gradient-Based One-Side Sampling`

Reply:
Resolved via 4bdcdd5